octobot_trading/exchanges/parser/orders_parser.py

```diff
@@ -0,0 +1,730 @@
+            enable_log=False,
+        )
+
+    def _parse_fees(self):
+        # optional
+        self._try_to_find_and_set(
+            OrderCols.FEE.value,
+            self.FEE_KEYS,
+            parse_method=self.found_fees,
+            enable_log=False,
+            use_info_sub_dict=self.USE_INFO_SUB_DICT_FOR_FEES,
+        )
+
+    def found_fees(self, fees_dict_or_list):
+        # fees example for paid fees in USDT:
+        # {'code': 'USDT', 'cost': -0.015922}
+        fees_dict = None
+        if type(fees_dict_or_list) is list:
+            fees_dict = fees_dict_or_list[0]
+        elif type(fees_dict_or_list) is dict:
+            fees_dict = fees_dict_or_list
+        if fees_dict:
+            if ExchangeConstantsFeesColumns.CURRENCY.value not in fees_dict and (
+                currency := fees_dict.get("code")
+            ):
+                fees_dict[ExchangeConstantsFeesColumns.CURRENCY.value] = currency
+                fees_dict.pop("code")
+            if fee := fees_dict[ExchangeConstantsFeesColumns.COST.value]:
+                fees_dict[ExchangeConstantsFeesColumns.COST.value] = (
+                    fee if fee < 0 else fee * -1
+                )
+            if (
+                fees_dict[ExchangeConstantsFeesColumns.CURRENCY.value]
+                and fees_dict[ExchangeConstantsFeesColumns.COST.value]
+            ):
+                return fees_dict
+
+    async def _fetch_if_missing(self):
+        to_find_id = self.formatted_record.get(OrderCols.ID.value)
+        if (
+            self.debugging_report_dict
+            and to_find_id
+            and (symbol := self.formatted_record.get(OrderCols.SYMBOL.value))
+        ):
+            if fetched_order := await self.exchange.get_order(
+                order_id=to_find_id, symbol=symbol, check_completeness=False
+            ):
+                self.fetched_order = fetched_order  # just for debugging purpose
+                # overwrite with fetched order details
+                for key in (
+                    OrderCols.STATUS.value,
+                    OrderCols.TIMESTAMP.value,
+                    OrderCols.SYMBOL.value,
+                    OrderCols.SIDE.value,
+                    OrderCols.OCTOBOT_ORDER_TYPE.value,
+                    OrderCols.TYPE.value,
+                    OrderCols.TAKER_OR_MAKER.value,
+                    OrderCols.PRICE.value,
+                    OrderCols.FILLED_PRICE.value,
+                    OrderCols.AVERAGE.value,
+                    OrderCols.AMOUNT.value,
+                    OrderCols.REMAINING.value,
+                    OrderCols.FILLED_AMOUNT.value,
+                    OrderCols.COST.value,
+                    OrderCols.REDUCE_ONLY.value,
+                    OrderCols.TAG.value,
+                    OrderCols.FEE.value,
+                ):
+                    self.set_fetched_attribute(fetched_order, key)
+
+    def set_fetched_attribute(self, fetched_order, key):
+        nothing = "nothing"
```

Comment:
?

Reply:
Actually that's not what we want, I removed it